src/content/keyAutoAdd/lib/keyAutoAdd.ts

```diff
@@ -45,6 +49,78 @@ export class KeyAutoAdd {
     );
   }
 
+  private setNotificationSize(size: 'notification' | 'fullscreen' | 'hidden') {
+    let styles: Partial<CSSStyleDeclaration>;
+    const defaultStyles: Partial<CSSStyleDeclaration> = {
+      outline: 'none',
+      border: 'none',
+      zIndex: '9999',
+      position: 'fixed',
+      top: '0',
+      left: '0',
+    };
+
+    if (size === 'notification') {
+      styles = {
+        width: '22rem',
+        height: '8rem',
+        position: 'fixed',
+        top: '1rem',
+        right: '1rem',
+        left: 'initial',
+        boxShadow: 'rgba(0, 0, 0, 0.1) 0px 0px 6px 3px',
+        borderRadius: '0.5rem',
+      };
+    } else if (size === 'fullscreen') {
+      styles = {
+        width: '100vw',
+        height: '100vh',
+      };
+    } else {
+      styles = {
+        width: '0',
+        height: '0',
+        position: 'absolute',
+      };
+    }
+
+    this.ui.style.cssText = '';
+    Object.assign(this.ui.style, defaultStyles);
+    Object.assign(this.ui.style, styles);
+
+    const iframeUrl = new URL(
+      browser.runtime.getURL('pages/progress-connect/index.html'),
+    );
+    const params = new URLSearchParams({ mode: size });
+    iframeUrl.hash = '?' + params.toString();
+    if (this.ui.src !== iframeUrl.href && size !== 'hidden') {
+      this.ui.src = iframeUrl.href;
+    }
+  }
+
+  private addNotification() {
+    const { resolve, reject, promise } = withResolvers<void>();
+    if (this.ui) {
+      resolve();
+      return promise;
+    }
+    const pageUrl = browser.runtime.getURL('pages/progress-connect/index.html');
+    const iframe = document.createElement('iframe');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.src = pageUrl;
+    document.body.appendChild(iframe);
+    iframe.addEventListener('load', () => {
+      resolve();
+      sleep(500).then(() =>
+        this.postMessage('PROGRESS', { steps: this.steps }),
+      );
+    });
+    iframe.addEventListener('error', reject, { once: true });
```

Comment:
Can't we register the events before appending the iframe to the body?

Reply:
No, the react app will listen to events only after load.